Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs

```diff
@@ -123,86 +152,99 @@ private void HandlePlacement()
-                    ParentGameObjectToPlace = gameObject.transform.parent.gameObject;
-                }
-            }
+            SpatialMappingManager.Instance.DrawVisualMeshes = IsBeingPlaced && AllowMeshVisualizationControl;
+        }
 
-            if (ParentGameObjectToPlace != null && !gameObject.transform.IsChildOf(ParentGameObjectToPlace.transform))
+        /// <summary>
+        /// If we're using the spatial mapping, check to see if we got a hit, else use the gaze position.
+        /// </summary>
+        /// <returns></returns>
+        private static Vector3 GetPlacementPosition(Vector3 headPosition, Vector3 gazeDirection, float defaultGazeDistance)
+        {
+            RaycastHit hitInfo;
+            if (SpatialMappingRaycast(headPosition, gazeDirection, out hitInfo))
             {
-                Debug.LogWarning("The specified parent object is not a parent of this object.");
+                return hitInfo.point;
             }
+            return GetGazePlacementPosition(headPosition, gazeDirection, defaultGazeDistance);
         }
 
-        private static void SetLayerRecursively(Transform objectToSet, bool useDefaultLayer)
+        /// <summary>
+        /// Does a raycast on the spatial mapping layer to try to find a hit.
+        /// </summary>
+        /// <param name="origin"></param>
+        /// <param name="direction"></param>
+        /// <param name="spatialMapHit"></param>
+        /// <returns>Wheter it found a hit or not</returns>
+        private static bool SpatialMappingRaycast(Vector3 origin, Vector3 direction, out RaycastHit spatialMapHit)
         {
-            if (useDefaultLayer)
+            if (SpatialMappingManager.Instance != null)
             {
-                int defaultLayerId;
-                if (defaultLayersCache.TryGetValue(objectToSet.gameObject, out defaultLayerId))
+                RaycastHit hitInfo;
+                if (Physics.Raycast(origin, direction, out hitInfo, 30.0f, SpatialMappingManager.Instance.LayerMask))
                 {
-                    objectToSet.gameObject.layer = defaultLayerId;
-                    defaultLayersCache.Remove(objectToSet.gameObject);
+                    spatialMapHit = hitInfo;
+                    return true;
                 }
             }
-            else
-            {
-                defaultLayersCache.Add(objectToSet.gameObject, objectToSet.gameObject.layer);
-
-                objectToSet.gameObject.layer = IgnoreRaycastLayer;
-            }
+            spatialMapHit = new RaycastHit();
+            return false;
+        }
 
-            for (int i = 0; i < objectToSet.childCount; i++)
+        /// <summary>
+        /// Get gaze position from the GazeManagers hit or just infront of the user
+        /// </summary>
+        /// <param name="headPosition"></param>
+        /// <param name="gazeDirection"></param>
+        /// <param name="defaultGazeDistance"></param>
+        /// <returns></returns>
```

Comment:
We should probably add the descriptions of these parameters and what we're returning.